src/lib.rs

```diff
@@ -71,16 +72,28 @@ pub fn init(config: &Config) {
 }
 
 #[no_mangle]
-pub fn start(_: &Lua, config: Config) -> LuaResult<()> {
+pub fn start(_: &Lua, config: Config) -> LuaResult<(bool, Option<String>)> {
```

Comment:
This is the first instance of how I am probably going to fix #179. Instead of relying on the `LuaResult`, return a boolean and an optional error message. The method can then simply be called via `assert(grpc.start())` to raise a Lua error (if the boolean is `false`).